addons/controllers/webhooks/webhooks.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2022 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package webhooks provides functions to manage webhook TLS certificates
+package webhooks
+
+import (
+	"context"
+	"time"
+
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/rest"
+
+	"github.com/vmware-tanzu/tanzu-framework/pkg/v1/webhooks"
+)
+
+type WebhookTLS struct {
+	Ctx           context.Context
+	K8sConfig     *rest.Config
+	CertPath      string
+	KeyPath       string
+	Name          string
+	ServiceName   string
+	LabelSelector string
+	Logger        logr.Logger
+	TLSSecret     *corev1.Secret
+	Namespace     string
+	InvalidWithIn time.Duration
+}
+
+func (w *WebhookTLS) Manage() error {
+	err := webhooks.ValidateTLSSecret(w.TLSSecret, w.InvalidWithIn)
+	if err != nil {
+		w.Logger.Info("Invalid webhook tls secret: " + err.Error())
+		w.Logger.Info("Installing new certificates")
+		w.TLSSecret, err = webhooks.InstallNewCertificates(w.Ctx, w.K8sConfig, w.CertPath, w.KeyPath, w.Name, w.Namespace, w.ServiceName, w.LabelSelector)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func ManageTLSCertificates(whtls *WebhookTLS, checkFrequency time.Duration) {
+	for {
+		select {
+		case <-whtls.Ctx.Done():
+			return
+		default:
```

Comment:
lets update to `time.After` and remove `time.Sleep`
```
 case <-time.After(checkFrequency):
```

Reply:
+1

Reply:
impleented ticker instead.